src/common/telemetry/telemetry-workflow.ts

```diff
@@ -32,44 +34,47 @@ export class TelemetryWorkflow extends Component {
       throw new Error('TelemetryWorkflow works only with instances of NodeProject.')
     }
 
+    const rcFilePath = ProjenrcFile.of(project)?.filePath
+
+    if (!rcFilePath) {
+      throw new Error('TelemetryWorkflow works only with projects that have a ProjenrcFile component.')
+    }
+
     super(project)
 
     const workingDirectory = options.outdir
-    const rcFile = ProjenrcFile.of(project)?.filePath
-    const paths = rcFile ? [rcFile] : undefined
     const nodeVersion = options.workflowNodeVersion ?? project.package.minNodeVersion
     const workflow = githubInstance.addWorkflow('telemetry')
-    workflow.on({pullRequest: {paths, types: ['opened', 'synchronize']}})
+    workflow.on({pullRequest: {paths: [rcFilePath], types: ['opened', 'synchronize']}})
 
-    const telemetryJob = runScriptJob({
-      command: 'default',
-      workingDirectory,
-      projectPackage: project.package,
-      runScriptCommand: project.runScriptCommand,
-      nodeVersion,
-    })
+    const env: Record<string, string> = {[telemetryEnableEnvVar]: '1'}
+    const {reportTargetAuthTokenVar} = options
 
-    workflow.addJob('telemetry', {...telemetryJob, env: {[telemetryEnableEnvVar]: '1'}})
+    if (reportTargetAuthTokenVar) {
+      env[reportTargetAuthToken] = `\${{ secrets.${reportTargetAuthTokenVar} }}`
+    }
+
+    workflow.addJob('telemetry', {
+      env,
+      ...runScriptJob({
+        command: 'default',
+        workingDirectory,
+        projectPackage: project.package,
+        runScriptCommand: project.runScriptCommand,
+        nodeVersion,
+      }),
+    })
   }
 
   /**
    * Creates a workflow within the given project
    * or within the project parent (for subprojects).
    */
   static addToProject(project: NodeProject, options: TelemetryWorkflowOptions) {
-    const {outdir, workflowNodeVersion} = options
```

Comment:
Also added a commit that removes any logic related to parent-subproject relation, since we aim for `projenrc` file that is only present on root project in a nested setup.